### ISSUE TITLE
Make the names of r2d generated images more readable

### ISSUE THREAD

**Issue by sgibson91** [jupyter/repo2docker#701]
### Proposed change

Hi! :wave:

I've always found the naming convention of images generated by repo2docker a little jarring, particularly the rogue `2d` that seems to crop up after hyphens (`-`) and slashes (`/`).

I propose to change this:
```
<some-prefix>-binder-2dexamples-2drequirements-55ab5c
```
to this:
```
<some-prefix>-binder-examples-requirements-55ab5c
```

### Alternative options

There isn't one really, I can just continue to cope 🙂

### Who would use this feature?

I'm unsure how repo2docker is utilised as a single resource. I think in terms of managing a BinderHub/JupyterHub, this would make the image registries and the user pod names more readable.

### How much effort will adding it take?

I hope only a small amount but I'm unfamiliar with r2d's architecture!

### Who can do this work?

Need to know how these names are constructed in r2d and understand what unicode/_something_ conversion is happening to cause the `2d`s.

**Comment by manics** [comment]
It looks like this is the line that escapes the name: https://github.com/jupyter/repo2docker/blob/f19e159dfe1006dbd82c7728e15cdd19751e8aec/repo2docker/app.py#L393-L395

And there's a configuration property:
https://github.com/jupyter/repo2docker/blob/f19e159dfe1006dbd82c7728e15cdd19751e8aec/repo2docker/app.py#L273-L281

This is just from looking through the code, I haven't checked whether it works.

**Comment by sgibson91** [comment]
Good place to start, thanks @manics!

**Comment by consideRatio** [comment]
Related discussion in Kubespawner: https://github.com/jupyterhub/kubespawner/issues/324

**Comment by manics** [comment]
A problem would be how to distinguish between `binder-examples/requirements` and `binder/examples-requirements` since both would end up as `binder-examples-requirements`. @sgibson91 Is this something you're still interested in solving?

**Comment by sgibson91** [comment]
I don't think this is a mission critical issue. I think if there's a sensible, low-friction path to addressing this while mitigating the problem you raised, then it's worth pursuing. Otherwise, I'm happy to not die on this hill 😄
